diesel_tests/tests/types.rs

```diff
@@ -1508,3 +1512,65 @@ fn cchar_to_sql() {
     assert!(query_to_sql_equality::<CChar, u8>(r#"0::"char""#, b'\0'));
     assert!(query_to_sql_equality::<CChar, u8>(r#"'0'::"char""#, b'0'));
 }
+
+#[cfg(feature = "postgres")]
+#[test]
+fn citext_fields() {
+    let connection = &mut connection();
+
+    // Enable the CIText extension
+    diesel::sql_query("CREATE EXTENSION IF NOT EXISTS citext")
+        .execute(connection)
+        .unwrap();
+
+    diesel::sql_query(
+        "CREATE TABLE case_insensitive (
+            id SERIAL PRIMARY KEY,
+            non_null_ci citext NOT NULL,
+            nullable_ci citext NULL,
+            null_value citext NULL)",
+    )
+    .execute(connection)
+    .unwrap();
+
+    table! {
+        case_insensitive (id) {
+            id -> Int4,
+            non_null_ci -> Citext,
+            nullable_ci -> Nullable<Citext>,
+            null_value -> Nullable<Citext>,
+        }
+    }
+
+    let rows_inserted = insert_into(case_insensitive::table)
+        .values((
+            case_insensitive::non_null_ci.eq("UPPERCASE_VALUE".to_string()),
+            case_insensitive::nullable_ci.eq("lowercase_value".to_string()),
```

Comment:
Maybe we can put not a string but an &str on this one so that both types are tested?

Reply:
Sounds like a good call :+1: